File: NEsper.Tests/support/core/SupportInternalEventRouter.cs
using System;
using System.Collections.Generic;

using net.esper.core;
using net.esper.events;

namespace net.esper.support.core
{

    public class SupportInternalEventRouter : InternalEventRouter
    {
        private IList<EventBean> routed = new List<EventBean>();

        public virtual void Route(EventBean _event)
        {
            routed.Add(_event);
        }

        public IList<EventBean> Routed
        {
            get { return routed; }
        }

        public IList<EventBean> GetRouted()
        {
            return routed;
        }

        public virtual void Reset()
        {
            routed.Clear();
        }
    }
}

File: trunk/esper/src/test/java/net/esper/support/core/SupportInternalEventRouter.java
package net.esper.support.core;

import net.esper.core.InternalEventRouter;
import net.esper.core.EPStatementHandle;
import net.esper.event.EventBean;

import java.util.List;
import java.util.LinkedList;

public class SupportInternalEventRouter implements InternalEventRouter
{
    private List<EventBean> routed  = new LinkedList<EventBean>();

    public void route(EventBean event, EPStatementHandle epStatementHandle)
    {
        routed.add(event);
    }

    public List<EventBean> getRouted()
    {
        return routed;
    }

    public void reset()
    {
        routed.clear();
    }
}
